File: 조준장/level3/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    answer = 0
    s_data =  sorted(data, key=lambda x: (x[col - 1], -x[0]))

    for i in range(row_begin-1, row_end):
        temp = 0
        for d in s_data[i]:
            temp += d % (i+1)
        answer ^= temp
        
    return answer

File: 이훈일/Programmers/Level2/테이블 해시 함수.java
import java.util.*;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        Arrays.sort(data, new Comparator<int[]>(){
            @Override
            public int compare(int[] i1, int[] i2) {
                if(i1[col-1] == i2[col-1]) {
                    return i2[0] - i1[0];
                }
                
                return i1[col-1] - i2[col-1];
            }
        });
        
        int result = 0;
        for(int i = row_begin-1; i < row_end; i++) {
            int mod = 0;
            
            for(int j = 0; j < data[i].length; j++) {
                mod += data[i][j] % (i+1);
            }
            
            result = result ^ mod;
        }
        
        return result;
    }
}
